ndcube/ndcube.py

```diff
@@ -918,6 +919,37 @@ def __neg__(self):
         return self._new_instance(data=-self.data)
 
     def __add__(self, value):
+        if isinstance(value, NDData) and value.wcs is None:
+            if self.unit is not None and value.unit is not None:
+                    value_data = value.data * value.unit.to(self.unit)
+            elif self.unit is None:
+                value_data = value.data
+            else:
+                raise TypeError("Cannot add unitless NDData to a unitful NDCube.")
+
+            # addition
+            new_data = self.data + value_data
```

Comment:
The addition should be done as part of the masked array addition.  You've already done this below, you just need to extract the added data from the results as well as the mask.